src/ncp-spinel/SpinelNCPInstance.cpp

```diff
@@ -1439,6 +1481,30 @@ SpinelNCPInstance::handle_ncp_spinel_value_is(spinel_prop_key_t key, const uint8
 			);
 		}
 
+	} else if ((key == SPINEL_PROP_THREAD_BA_PROXY_STREAM)) {
+		const uint8_t* frame_ptr(NULL);
+		unsigned int frame_len(0);
+		uint16_t locator;
+		uint16_t port;
+		spinel_ssize_t ret;
+
+		ret = spinel_datatype_unpack(
+			value_data_ptr,
+			value_data_len,
+			SPINEL_DATATYPE_DATA_S SPINEL_DATATYPE_UINT16_S SPINEL_DATATYPE_UINT16_S,
+			&frame_ptr,
+			&frame_len,
+			&locator,
+			&port
+		);
+
+		__ASSERT_MACROS_check(ret > 0);
+
+		// Analyze the packet to determine if it should be dropped.
+		if ((ret > 0)) {
+			signal_border_agent_proxy_stream(frame_ptr, frame_len, locator, port);
```

Comment:
Yes, similar to how we do for spinel. I would personally put the locator and port as the first four bytes, and then have the coap frame be the remaining bytes.